### PR TITLE
Appointment search api

### PR DESCRIPTION
**Pull request by MonishaSivanathan**
- Added search_by_patient_name and date_filter
- added exception handling and logger

### CODE REVIEW

File: controller/appointment_controller.py [hunk 1]
```diff
@@ -213,3 +214,56 @@ def get_by_id(
...
+        try:
+            responce_name = AidboxApi.make_request(method="GET", endpoint=f"/$query/appointmentByStartDate?start={state_date}&end={end_date}")
+            data = responce_name.json()
+            if not data.get('data', []):
+                return JSONResponse(
+                    content=[],
+                    status_code=status.HTTP_404_NOT_FOUND
+                )
+            return JSONResponse(
+                content=data,
```
Comment (tamsud) [inline controller/appointment_controller.py:254]: can you return the object same as get_by_id of appointment and get_by_all appointment?

File: controller/appointment_controller.py [hunk 2]
```diff
@@ -213,3 +214,56 @@ def get_by_id(
...
+        try:
+            responce_name = AidboxApi.make_request(method="GET", endpoint=f"/$query/appointmentByPatientName?patientName={name}")
+            data = responce_name.json()
+            if not data.get('data', []):
+                return JSONResponse(
+                    content=[],
+                    status_code=status.HTTP_404_NOT_FOUND
+                )
+            return JSONResponse(
+                content=data,
```
Comment (tamsud) [inline controller/appointment_controller.py:229]: can you return the object same as get_by_id of appointment and get_by_all appointment?

File: controller/master_controller.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+from fastapi import status, HTTPException
```
Comment (tamsud) [inline controller/master_controller.py:1]: why this pr has master details?

**Review by tamsud — CHANGES_REQUESTED** [review]

File: controller/appointment_controller.py (hunk 1 above)
Comment (MonishaSivanathan) [reply controller/appointment_controller.py]: returned

File: controller/appointment_controller.py (hunk 2 above)
Comment (MonishaSivanathan) [reply controller/appointment_controller.py]: returned

File: controller/appointment_controller.py [hunk 4]
```diff
@@ -188,8 +189,7 @@ def get_by_id(
             ):
                 logger.info("No Record Found")
                 error_response_data = {
-                "error": "Unable to retrieve datas",
-                "details": str(e),
+                "error": "Unable to retrieve datas"
```
Comment (tamsud) [inline controller/appointment_controller.py]: give proper msg

**Review by tamsud — APPROVED** [review]

**Pull request merged**